plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderRegistry.java

```diff
@@ -122,14 +122,20 @@ private void loadExtensions(IExtensionRegistry registry) {
         {
             IConfigurationElement[] extElements = registry.getConfigurationElementsFor(DataSourceProviderDescriptor.EXTENSION_ID);
             // Sort - parse providers with parent in the end
-            Arrays.sort(extElements, (o1, o2) -> {
-                String p1 = o1.getAttribute(RegistryConstants.ATTR_PARENT);
-                String p2 = o2.getAttribute(RegistryConstants.ATTR_PARENT);
-                if (CommonUtils.equalObjects(p1, p2)) return 0;
-                if (p1 == null) return -1;
-                if (p2 == null) return 1;
-                return 0;
-            });
+            Arrays.sort(
+                extElements,
+                Comparator.comparingInt(e -> {
+                    boolean hasParent = !CommonUtils.isEmpty(e.getAttribute(RegistryConstants.ATTR_PARENT));
+                    if (!hasParent) {
```

Comment:
Double negation. I also suggest extracting `e.getAttribute(RegistryConstants.ATTR_PARENT)` to a variable to avoid extra lookup

Reply:
Double negation is reasonable here. It makes it easy to read.

Reply:
I don't see how it's clearler than

```suggestion
                    String parent = e.getAttribute(RegistryConstants.ATTR_PARENT);
                    if (CommonUtils.isEmpty(parent)) {
```